detector/detector_test.go

```diff
@@ -0,0 +1,235 @@
+package detector
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/detect-angular-dashboards/api/gcom"
+	"github.com/grafana/detect-angular-dashboards/api/grafana"
+	"github.com/grafana/detect-angular-dashboards/logger"
+	"github.com/grafana/detect-angular-dashboards/output"
+)
+
+func TestDetector(t *testing.T) {
+	t.Run("meta", func(t *testing.T) {
+		cl := NewTestAPIClient(filepath.Join("testdata", "dashboards", "graph-old.json"))
+		d := NewDetector(logger.NewLeveledLogger(false), cl, gcom.NewAPIClient())
+		out, err := d.Run(context.Background())
+		require.NoError(t, err)
+		require.Len(t, out, 1)
+		require.Equal(t, "test case dashboard", out[0].Title)
+		require.Equal(t, "test case folder", out[0].Folder)
+		require.Equal(t, "d/test-case-dashboard/test-case-dashboard", out[0].URL)
+		require.Equal(t, "admin", out[0].CreatedBy)
+		require.Equal(t, "admin", out[0].UpdatedBy)
+		require.Equal(t, "2023-11-07T11:13:24+01:00", out[0].Created)
+		require.Equal(t, "2024-02-21T13:09:27+01:00", out[0].Updated)
+
+	})
+
+	t.Run("legacy panel", func(t *testing.T) {
```

Comment:
Should you just combine these tests maybe?

Reply:
Thanks, done!